packages/flutter_tools/lib/src/project.dart

```diff
@@ -260,6 +260,9 @@ abstract class XcodeBasedProject {
   /// The Flutter-managed Xcode config file for [mode].
   File xcodeConfigFor(String mode);
 
+  /// The script to run as part of the module pod build phase.
+  File get moduleBuildPhaseScript;
```

Comment:
I think we can use this in both projects.  In my opinion we shouldn't be polluting either Xcode projects build configs with environment variables that the scripts need, like FLUTTER_* or LOCAL_ENGINE.  If we use this script we can remove the Generated.xcconfig-type files.

Reply:
ah ok, so you're saying you envision everyone (project 1 and 2) to use this script. In the full-app templates, you would just make an xcode project that doesn't leak variables anywhere and that just uses this to make the re-entrant flutter tool calls? That would be awesome.

Reply:
I think I'm going to change it to just be the export environment variables so any scripts can be run after, whether it's xcode_backend or some future `flutter build` thing.

Reply:
I think they way you had it is great. Did it break things?

Reply:
Instead of this build phase being:
```
"${SRCROOT}/flutterapp/.ios/Flutter/flutter_build_phase.sh"
```
If it's:
```
"${SRCROOT}/flutterapp/.ios/Flutter/flutter_export_environment_variables.sh"
"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build
```
Then we can re-use the environment variables script in other places, like in the Runner app:
```
"${SRCROOT}/flutterapp/.ios/Flutter/flutter_export_environment_variables.sh"
"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" build
"$FLUTTER_ROOT/packages/flutter_tools/bin/xcode_backend.sh" embed
```

Reply:
Reusing this generated function for full-app cases makes sense. Though I think the full-apps don't run xcode_backend.sh embed. Since we own the "outer xcode project" in that case, we just always hard linked the framework in the .xcodeproj.